pkg/sm/sm.go

```diff
@@ -165,7 +166,9 @@ func New(ctx context.Context, cancel context.CancelFunc, e env.Environment, cfg
 	}
 
 	operationMaintainer := operations.NewMaintainer(ctx, interceptableRepository, postgresLockerCreatorFunc, cfg.Operations, waitGroup)
-	osbClientProvider := osb.NewBrokerClientProvider(cfg.HTTPClient.SkipSSLValidation, int(cfg.HTTPClient.ResponseHeaderTimeout.Seconds()))
+	osbClientTimeout := math.Min(float64(cfg.HTTPClient.Timeout), float64(cfg.Server.RequestTimeout))
```

Comment:
lets say server request timeout is picked here and timeout handler just expired and wrote, will this try error happening within a request flow also try writing? what about the other way around? is this ok ?

Reply:
It won't fail, because the ResponseWriter in our handler is replaced, by another writer, which if has already timedOut, will not attempt to write to the socket. See the implementation of TimeoutHandler.  I can still try it of course